### PR TITLE
React Aria Button on Autocomplete and Filters

### PR DESCRIPTION
**Pull request by joel-jeremy**

### CODE REVIEW

File: packages/desktop-client/e2e/transactions.test.js-snapshots/Transactions-filters-transactions-by-category-1-chromium-linux.png
Comment (joel-jeremy) [inline packages/desktop-client/e2e/transactions.test.js-snapshots/Transactions-filters-transactions-by-category-1-chromium-linux.png:1]: These are intentional. We use `theme.pillBackgroundSelected` throughout the app as background colors for pills so doing the same here.

**Comment by MatissJanis** [comment]
Happy to review this, but it looks like e2e tests are still failing.

**Comment by joel-jeremy** [comment]
@MatissJanis I'm encountering a weird issue where in the VRT tests, when clicking the Add button on the new transaction, it won't add the transaction. But when I open in a browser and do the same it does add the transaction. Any ideas?

**Comment by MatissJanis** [comment]
I think the problem is with creating transfers. When I click "create transfer" (inside account selection dropdown) - the popover just closes.

**Comment by joel-jeremy** [comment]
@MatissJanis The problem is the Notes filter. What's weird is it works when I manually test it. In playwright, when the Apply button is clicked it's not reflecting the "Test Acc Transfer" that was types in the input field.

**Comment by MatissJanis** [comment]
So there might be multiple problems with then PR then. This is the one I'm seeing:


https://github.com/actualbudget/actual/assets/886567/50db3ef3-0185-4cc4-bb60-7702220766ef

**Comment by joel-jeremy** [comment]
Looks like there is something with the interaction between the autocomplete Popover and the "Make Transfer"/"Manage Payees" Button inside the Popover. It could be treating the the click as a "click outside" - not exacrly sure what is happening. Any ideas @MatissJanis? I'll move to migrate other parts of the code first and do the autocomplete and filters migration last.

**Comment by MatissJanis** [comment]
Ok, so looked into this more. What happens is:

1. the autocomplete is opened when the input field is focused
2. when a button inside it (i.e. "make transfer") is clicked - the input field loses focus
3. losing focus triggers the "close" event
4. the `onPress` is never triggered for the button because at this point the component is no longer rendered

AFAIK there is no way to control the focus logic for the react-aria button component. So my only other option is to use a link here (`<a>`) that is styled to look and behave like a button. That _might_ solve the problem.

**Pull request closed without merging**